src/MonitorProfiler/MainProfiler/ExceptionTracker.cpp

```diff
@@ -0,0 +1,301 @@
+    BYTE context[],
+    void* clientData)
+{
+    // Callback is only used to get the function ID of the top frame.
+    *static_cast<FunctionID*>(clientData) = funcId;
+
+    // Cancel stack snapshot callbacks after the top frame.
+    return S_FALSE;
+}
+
+HRESULT ExceptionTracker::GetFullyQualifiedMethodName(FunctionID functionId, tstring& fullMethodName)
+{
+    HRESULT hr = S_OK;
+
+    ClassID classId;
+    ModuleID moduleId;
+    mdToken token;
+    IfFailLogRet(m_pCorProfilerInfo->GetFunctionInfo(
+        functionId,
+        &classId,
+        &moduleId,
+        &token
+    ));
+
+    if (0 == (token & mdtMethodDef))
+    {
+        return E_FAIL;
+    }
+    mdMethodDef methodDef = token;
+
+    ComPtr<IMetaDataImport2> pMetadataImport;
+    IfFailLogRet(m_pCorProfilerInfo->GetModuleMetaData(
+        moduleId,
+        CorOpenFlags::ofRead,
+        IID_IMetaDataImport2,
+        (IUnknown**)&pMetadataImport
+    ));
+
+    // Get Module Name
+    ULONG count = 0;
+    IfFailLogRet(m_pCorProfilerInfo->GetModuleInfo(
+        moduleId,
+        nullptr,
+        0,
+        &count,
+        nullptr,
+        nullptr
+    ));
+
+    unique_ptr<WCHAR[]> pModulePath(new (nothrow) WCHAR[count + 1]);
+    IfFailLogRet(m_pCorProfilerInfo->GetModuleInfo(
+        moduleId,
+        nullptr,
+        count + 1,
+        nullptr,
+        pModulePath.get(),
+        nullptr
+    ));
+
+    // Get Class Name
+    tstring className;
+    if (0 == classId)
+    {
+        // Probably a value type e.g. struct
+        className.assign(_T("[Unknown]"));
+    }
+    else
+    {
+        mdTypeDef typeDef = mdTokenNil;
+        IfFailLogRet(m_pCorProfilerInfo->GetClassIDInfo(
+            classId,
+            nullptr,
+            &typeDef
+        ));
+
+        IfFailLogRet(pMetadataImport->GetTypeDefProps(
+            typeDef,
+            nullptr,
+            0,
+            &count,
+            nullptr,
+            nullptr
+        ));
+
+        unique_ptr<WCHAR[]> pClassName(new (nothrow) WCHAR[count + 1]);
+        IfFailLogRet(pMetadataImport->GetTypeDefProps(
+            typeDef,
+            pClassName.get(),
+            count + 1,
+            nullptr,
+            nullptr,
+            nullptr
+        ));
+
+        className.assign(pClassName.get());
+    }
+
+    // Get Method Name
+    IfFailLogRet(pMetadataImport->GetMethodProps(
+        methodDef,
+        nullptr,
+        nullptr,
+        0,
+        &count,
+        nullptr,
+        nullptr,
+        nullptr,
+        nullptr,
+        nullptr
+    ));
+
+    unique_ptr<WCHAR[]> pMethodName(new (nothrow) WCHAR[count + 1]);
+    IfFailLogRet(pMetadataImport->GetMethodProps(
+        methodDef,
+        nullptr,
+        pMethodName.get(),
+        count + 1,
+        nullptr,
+        nullptr,
+        nullptr,
+        nullptr,
+        nullptr,
+        nullptr
+    ));
+
+    tstring modulePath(pModulePath.get());
+
+    fullMethodName.clear();
```

Comment:
Maybe add a comment showing what an expected `fullMethodName` is supposed to look like to help visually double check that this constructs it correctly?